vendors/circom/prover_main.cc

```diff
@@ -138,24 +139,52 @@ int RealMain(int argc, char** argv) {
       .set_help(
           "Verify the proof. By default verify is disabled. Use this flag "
           "to verify the proof with the public inputs.");
+ parser.AddFlag<base::Flag<size_t>>(&num_runs)
+    .set_short_name("-n")
+    .set_long_name("--num_runs")
+    .set_required()
+    .set_help("The number of times to run the proof generation");
 
   std::string error;
   if (!parser.Parse(argc, argv, &error)) {
     tachyon_cerr << error << std::endl;
     return 1;
   }
+  if (num_runs == 0) {
+    tachyon_cerr << "num_runs should be positive" << std::endl;
+    return 1;
+  }
 
-  switch (curve) {
-    case Curve::kBN254:
-      circom::CreateProof<math::bn254::BN254Curve>(
-          zkey_path, witness_path, proof_path, public_path, no_zk, verify);
-      break;
-    case Curve::kBLS12_381:
-      circom::CreateProof<math::bls12_381::BLS12_381Curve>(
-          zkey_path, witness_path, proof_path, public_path, no_zk, verify);
-      break;
+  
+
+    double total_time = 0.0;
+  double max_time = 0.0;
+  for (size_t i = 0; i < num_runs; ++i) {
+    auto start = std::chrono::high_resolution_clock::now();
+    switch (curve) {
+      case Curve::kBN254:
+        circom::CreateProof<math::bn254::BN254Curve>(
+            zkey_path, witness_path, proof_path, public_path, no_zk, verify);
+        break;
+      case Curve::kBLS12_381:
+        circom::CreateProof<math::bls12_381::BLS12_381Curve>(
+            zkey_path, witness_path, proof_path, public_path, no_zk, verify);
+        break;
+    }
+    auto end = std::chrono::high_resolution_clock::now();
+    std::chrono::duration<double> elapsed = end - start;
+    double time_taken = elapsed.count();
+    total_time += time_taken;
+    max_time = std::max(max_time, time_taken);
+    std::cout << "Run " << (i + 1) << ", Time Taken: " << time_taken << " seconds" << std::endl;
```

Comment:
Since `base::TimeDelta` will be printed with a suffix "s", so please remove " seconds".

```suggestion
    std::cout << "Run " << (i + 1) << ", Time Taken: " << time_taken << std::endl;
```

Reply:
implemented these changes in the current commit